### ISSUE TITLE
Provide LMCSS ligand file for ligand and receptor prep

### ISSUE THREAD

**Issue by j-wags** [drugdata/D3R#178]

**Comment by dkoes** [comment]
Related - I would like an easy way to reference back to the full set of provided files at any stage, for example by setting a 'challenge_data_dir' attribute in the targ_info_dict

**Comment by dkoes** [comment]
I decided to only use the cognate ligand in each reference receptor rather than create a consensus ligand to define the box, so I don't need access to the full challenge data (apo structure uses a box around the center).  I think it is weird that that targ_info_dict has everything in it - I had to parse the receptor name to get the appropriate prefix (e.g. hiTanimoto) in order to figure out how to index into targ_info_dict to extract the appropriate lig_name.  It also isn't clear to me why there are lists for somethings.

I am doing "scientific" prep in the technical prep since technical prep lets me return multiple things (the receptor and the ligand file in this case) and scientific prep doesn't.  You might consider implementing a more general means of communicating between the phases (e.g., json) to provide more flexibility to workflows (or not).

Here's my code in case you are interested: https://github.com/dkoes/gnina-celpp

Unrelated: Replacing `logging.info(sys.exc_info())` with `logging.info(traceback.format_exc())` in the `except` blocks really aids debugging.

**Comment by dkoes** [comment]
Yes, the (or not) was included because allowing more flexibility reduces modularity.  

The sample autodock vina scripts remove the ligand in scientific prep.  Since I don't, trying to mix workflows wouldn't work.  

You either need to have a more defined api where you say exactly what files of what type are generated by preparation (in which case I'd request that in addition to the receptor you allow the cognate ligand and possibly cofactors/waters be propagated), or you need to give up on modularity and provide the most flexible interface.

A hybrid approach is to have a json specification for passing data between the passes.  The json object has some standard fields (e.g., receptor in pdb format, cognate ligand in pdb format), but users can create additional fields.  If only the standard fields are set, then it is assumed the modules are "plug and play".
